### DISCUSSION TITLE
ScriptReference/MonoBehaviour.OnCollisionStay

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#493] [Notes]
# ScriptReference/MonoBehaviour.OnCollisionStay

https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnCollisionStay.html

**Comment by jschiff** [comment]
> Sent each frame where a collider on another object is touching this object's collider 
This statement is a bit misleading. This callback actually happens at the end of each _physics simulation_, not each frame. In normal circumstances this is at the cadence of [Time.fixedDeltaTime](https://docs.unity3d.com/ScriptReference/Time-fixedDeltaTime.html).

A common pitfall with OnCollisionStay is to read input in the method, which cannot work reliably due to it not actually running every frame.
